### ISSUE TITLE
Standardise sorting and quoting of translation systems

### ISSUE THREAD

**Issue by emteknetnz** [silverstripe/.github#63]
Both collecting/tx-pushing with i18nTextCollector (now run by tx-translator) and tx pulling with silverstripe/tx-translator (formally cow) which runs `tx pull` both seem to have differences in:
- sorting keys
- quoting strings

This leads to numerous [issues](https://github.com/silverstripe/.github/issues/64) during merge-ups of duplicated keys and strings quoted differently, making things far more complicated than they should be

There should be a shared method or set of methods in framework that alpha sort keys and ensure that quotes are applied consistently

~~For the alpha key sorting, since other projects may use the i18nTextCollector, it should be an opt-in operation~~ Actually it's fine to reorder them since it's only outputted to yml files, it's not a real behavioral change

## Acceptance criteria
- Translation Keys are always in the same order.
- Translation values use systematic quoting styles.

## PRs
- https://github.com/silverstripe/silverstripe-tx-translator/pull/15

**Comment by emteknetnz** [comment]
### tl;dr

- Update tx-translator to key sort
- Quote seems consistent between the systems. Leave for now, if it turns out my investigation was wrong we can fix it later

### Some investigation

tx-translator TX_PULL
- **won't keys reorder**
- will single quote strings with spaces in them
- will not quote strings that do not need to be quoted
- will single quote string with double single quote for a ' in the string - https://github.com/tractorcow-farm/silverstripe-fluent/pull/802/files#diff-0df2677d84007a9e9dbf1c06cbc8ee7cc2ceb663efb442203e5206de2efcdf32R53
- js/json is done with double quotes - https://github.com/symbiote/silverstripe-gridfieldextensions/pull/365/files
- js got key sorted - https://github.com/silverstripe/silverstripe-mfa/pull/499/files#diff

i18nTextCollectorTask
- **will reorder alpha** - uses ksort() in several places in i18nTextCollectorTask
- does single quote string for things with spaces in them
- will do single quote string with double single quote for a ' in the string
- will do single quote string with unquoted double quot for a " in the string
- will not quote strings that do not need to be quoted
- I don't believe that textcollector currently text collects

**Comment by GuySartorelli** [comment]
@emteknetnz I can't see an associated PR - what needs peer review for this issue?

**Comment by emteknetnz** [comment]
My bad have added

**Comment by GuySartorelli** [comment]
PR merged, no outstanding acceptance criteria
